SoC-2019-Microprojects.md

```diff
@@ -218,6 +218,28 @@ that we have `struct dir_iterator` (see dir-iterator.h), convert these
 to use the dir-iterator to simplify the code. Do only one conversion
 per microproject.
 
+### Let `git fetch <remote> HEAD` use the current branch name
+
+When fetching from a remote, one can specify which ref(s) to fetch.
+There is no convenient way to specify the current branch, though, and
+`HEAD` may be a good shortcut. Or `@`. Or `-`.
+
+### Teach `git range-diff` new tricks
+
+When contributing patches to Git, developers often need to compare
+in what form those patches made it upstream (e.g. to pick up fixes
+made by the Git maintainer in preparation for a new iteration). In
+such a case, a `--left-only` (or `--right-only`) option would come
+in real handy, to allow `git range-diff --left-only <branch>..upstream/pu`
+to skip commits that are upstream only (as there will be a metric
+ton of them).
```

Comment:
I guess `git range-diff --right-only <branch>..upstream/pu` would skip commits that are only in <branch>. I wonder if the syntax could be `git range-diff --left-only <branch> upstream/pu` (so without `..`) though. I also wonder if  `git range-diff --left-only <branch>...upstream/pu` (so with 3 dots) or `git range-diff --left-only <base> <rev1> <rev2>` would make sense?

Reply:
The way `git range-diff` parses its option, all three invocations (1, 2 and 3 arguments all are translated into the two arguments `A..B C..D`) boil down to the same. So yes, it would work. It would actually be hard to make it *not* work for all three invocations.

Reply:
Ok, nice!